uniffi_bindgen/src/bindings/kotlin/templates/RustBufferHelpers.kt

```diff
@@ -282,6 +282,70 @@ internal fun String.write(buf: RustBufferBuilder) {
     buf.put(byteArr)
 }
 
+{% when Type::Timestamp -%}
+{% let type_name = typ|type_kt %}
+
+internal fun lift{{ canonical_type_name }}(rbuf: RustBuffer.ByValue): {{ type_name }} {
+    return liftFromRustBuffer(rbuf) { buf ->
+        read{{ canonical_type_name }}(buf)
+    }
+}
+
+internal fun read{{ canonical_type_name }}(buf: ByteBuffer): {{ type_name }} {
+    val seconds = buf.getLong()
+    val nanoseconds = buf.getInt().toLong()
```

Comment:
Ah, I see, so Java uses signed integers here while Rust uses unsigned. That's unfortunately, but par for the course here with the Kotlin bindings :-/

Could you please add a comment here noting that the data was written to the buffer as an unsigned integer, but we're deliberately reading as a signed integer because that's what Java/Kotlin need? Just to make it clear that we're doing it on purpose, because we've had bugs in this code in the past where we were doing it by accident.

Alternately - what do you think about enforcing this at the Rust layer, so that all consumers get consistent behaviour? For example, our String type [limits everyone to an i32 string length](https://github.com/mozilla/uniffi-rs/blob/b5a54b162c1af4989c9092761d211f5b942c0ded/uniffi/src/lib.rs#L313) because Java/Kotlin strings have this limit, to ensure consistency for all consumers. In practice that would mean serializing durations as signed rather than unsigned integers, similar to what we do for string length there.